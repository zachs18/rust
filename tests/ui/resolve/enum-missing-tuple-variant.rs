enum Foo {
    //~^ HELP consider importing this tuple variant
    A(u32),
    B(u32),
}

fn main() {
    let _: Foo = Foo(0);
    //~^ ERROR expected function
    //~| HELP try to construct one of the enum's variants

    let s: Foo = Foo.A(0);
    //~^ ERROR expected value, found enum `Foo`
    //~| HELP use the path separator to refer to a variant

    let s: Foo = Foo.C(0);
    //~^ ERROR expected value, found enum `Foo`
    //~| HELP the following enum variants are available

    match s {
        A(..) => {}
        //~^ ERROR cannot find tuple struct or tuple variant `A` in this scope
        Foo(..) => {}
        //~^ ERROR expected tuple struct or tuple variant
        //~| HELP try to match against one of the enum's variants
        _ => {}
    }
}
