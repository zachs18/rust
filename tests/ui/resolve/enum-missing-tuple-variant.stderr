error[E0423]: expected value, found enum `Foo`
  --> $DIR/enum-missing-tuple-variant.rs:12:18
   |
LL |     let s: Foo = Foo.A(0);
   |                  ^^^
   |
note: the enum is defined here
  --> $DIR/enum-missing-tuple-variant.rs:1:1
   |
LL | / enum Foo {
LL | |
LL | |     A(u32),
LL | |     B(u32),
LL | | }
   | |_^
help: use the path separator to refer to a variant
   |
LL |     let s: Foo = Foo::A(0);
   |                     ~~

error[E0423]: expected value, found enum `Foo`
  --> $DIR/enum-missing-tuple-variant.rs:16:18
   |
LL |     let s: Foo = Foo.C(0);
   |                  ^^^
   |
note: the enum is defined here
  --> $DIR/enum-missing-tuple-variant.rs:1:1
   |
LL | / enum Foo {
LL | |
LL | |     A(u32),
LL | |     B(u32),
LL | | }
   | |_^
help: the following enum variants are available
   |
LL |     let s: Foo = (Foo::A(/* fields */)).C(0);
   |                  ~~~~~~~~~~~~~~~~~~~~~~
LL |     let s: Foo = (Foo::B(/* fields */)).C(0);
   |                  ~~~~~~~~~~~~~~~~~~~~~~

error[E0531]: cannot find tuple struct or tuple variant `A` in this scope
  --> $DIR/enum-missing-tuple-variant.rs:21:9
   |
LL |         A(..) => {}
   |         ^ not found in this scope
   |
help: consider importing this tuple variant
   |
LL + use Foo::A;
   |

error[E0532]: expected tuple struct or tuple variant, found enum `Foo`
  --> $DIR/enum-missing-tuple-variant.rs:23:9
   |
LL |         Foo(..) => {}
   |         ^^^
   |
note: the enum is defined here
  --> $DIR/enum-missing-tuple-variant.rs:1:1
   |
LL | / enum Foo {
LL | |
LL | |     A(u32),
LL | |     B(u32),
LL | | }
   | |_^
help: try to match against one of the enum's variants
   |
LL |         Foo::A(..) => {}
   |         ~~~~~~
LL |         Foo::B(..) => {}
   |         ~~~~~~

error[E0423]: expected function, tuple struct or tuple variant, found enum `Foo`
  --> $DIR/enum-missing-tuple-variant.rs:8:18
   |
LL |     let _: Foo = Foo(0);
   |                  ^^^
   |
note: the enum is defined here
  --> $DIR/enum-missing-tuple-variant.rs:1:1
   |
LL | / enum Foo {
LL | |
LL | |     A(u32),
LL | |     B(u32),
LL | | }
   | |_^
help: try to construct one of the enum's variants
   |
LL |     let _: Foo = Foo::A(0);
   |                  ~~~~~~
LL |     let _: Foo = Foo::B(0);
   |                  ~~~~~~

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0423, E0531, E0532.
For more information about an error, try `rustc --explain E0423`.
